File: app/routers/HelloWorldRouter.scala
package routers

import akka.stream.Materializer
import example.myapp.helloworld.grpc.{HelloReply, HelloRequest}
import javax.inject.Inject

import scala.concurrent.Future

class HelloWorldRouter @Inject() (mat: Materializer) extends example.myapp.helloworld.grpc.AbstractGreeterServiceRouter(mat) {
  override def sayHello(in: HelloRequest): Future[HelloReply] =
    Future.successful(HelloReply(s"Hello, ${in.name}!"))
}

File: app/routers/HelloWorldRouter.java
package routers;

import akka.actor.ActorSystem;
import akka.stream.Materializer;
import example.myapp.helloworld.grpc.HelloReply;
import example.myapp.helloworld.grpc.HelloRequest;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

@Singleton
public class HelloWorldRouter extends example.myapp.helloworld.grpc.AbstractGreeterServiceRouter {

  @Inject
  public HelloWorldRouter(Materializer mat, ActorSystem system) {
    super(mat, system);
  }

  @Override
  public CompletionStage<HelloReply> sayHello(HelloRequest in) {
    String greeting = String.format("Hello %s!", in.getName());
    return CompletableFuture.completedFuture(HelloReply.newBuilder().setMessage(greeting).build());
  }

}
